CODE_STYLE.md

```diff
@@ -0,0 +1,3293 @@
+# Prelude
+
+This guide is intended to become a reference for logstash developers, a central place where contributor can refer when in doubt about how to style their code. Nevertheless an style guide is also useful to make the code look similar, reducing complexities and especially when groups of contributors are growing.
+
+This style guide is based on the [guide][rails-style-guide] created by Bozhidar Batsov for [Rubocop][RuboCop], The Airbnb [guide][Airbnb], the Logstash team inputs, and others. There is also a rubocop file used to check this guide automatically.
+
+## Table of Contents
+
+* [Source Code Layout](#source-code-layout)
+* [Syntax](#syntax)
+* [Naming](#naming)
+* [Comments](#comments)
+  * [Comment Annotations](#comment-annotations)
+* [Classes](#classes--modules)
+* [Exceptions](#exceptions)
+* [Collections](#collections)
+* [Strings](#strings)
+* [Regular Expressions](#regular-expressions)
+* [Percent Literals](#percent-literals)
+* [Metaprogramming](#metaprogramming)
+
+## Source Code Layout
+
+  * Use `UTF-8` as the source file encoding. If you're using a ruby version older than 2.0, adding ```# encoding: utf-8``` on top of a ruby file will tell the compiler to use this encoding by default. If using a ruby VM older than 2.0, `UTF-8` is the default encoding.
```

Comment:
> If using a ruby VM older than 2.0, `UTF-8` is the default encoding

I think "older" should be "newer". In Ruby 2.0+, the default encoding is UTF-8. Prior in 1.8 and 1.9, I think the default encoding was assumed to be based on $LANG but overridable with the `# encoding:xyz` comment at the top of the file.